server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultDeleteTask.java

```diff
@@ -0,0 +1,65 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.vault.routes;
+
+import java.util.Optional;
+
+import org.apache.james.core.User;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.vault.DeletedMessageVault;
+
+import reactor.core.publisher.Mono;
+
+public class DeletedMessagesVaultDeleteTask implements Task {
+
+    static final String TYPE = "deletedMessages/delete";
+
+    private final DeletedMessageVault vault;
+    private final User user;
+    private final MessageId messageId;
+
+    DeletedMessagesVaultDeleteTask(DeletedMessageVault vault, User user, MessageId messageId) {
+        this.vault = vault;
+        this.user = user;
+        this.messageId = messageId;
+    }
+
+    @Override
+    public Result run() {
+        return Mono.from(vault.delete(user, messageId))
+            .doOnError(e -> LOGGER.error("Error while deleting message in DeletedMessageVault: {}", e.getMessage()))
```

Comment:
Please don't loose the stacktrace

Reply:
you can log information about `user` & `messageId` here

Reply:
```
LOGGER.error("Error while deleting message {} of user {} in DeletedMessageVault", messageId.serialize(), user.asString(), e)
```